test/bats/apigateway/apigateway.bats

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env bats
+
+# tags: apigateway, api-gateway, route, gateway, customDomains, upstreams
+# nume de fisiere only sau directoare
+
+BATS_LIBS_PATH="${LIBS_PATH:-../libs}" # fallback to relative path if not set
+load "${BATS_LIBS_PATH}/bats-assert/load"
+load "${BATS_LIBS_PATH}/bats-support/load"
+load '../setup.bats'
+
+setup_file() {
+    rm -rf /tmp/bats_test
+    mkdir -p /tmp/bats_test
+
+    uuid_v4_regex='^[0-9a-f]{8}-[0-9a-f]{4}-[4][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$'
+    # verifica daca este ipv4
+}
+
+setup() {
+    if [[ -f /tmp/bats_test/token ]]; then
+        export IONOS_TOKEN="$(cat /tmp/bats_test/token)"
+    fi
+}
+
+#content
+@test "Create ApiGateway Gateway" {
+    gateway_name="cli-test-gateway"
+    run ionosctl apigateway gateway create --name "$gateway_name" -o json 2> /dev/null
+    assert_success
+
+    gateway_id=$(echo "$output" | jq -r '.id')
+
+    assert_output -p "\"name\": \"cli-test-gateway\""
+    assert_output -p "\"status\": \"PROVISIONING\""
+
+    echo "created apigateway gateway $gateway_id ($gateway_name)"
+    echo "$gateway_id" > /tmp/bats_test/gateway_id
+    echo "$gateway_name" > /tmp/bats_test/gateway_name
+}
+
+@test "List ApiGateway Gateways" {
+    gateway_id=$(cat /tmp/bats_test/gateway_id)
+
+    # List Gateway (JSON output)
+    run ionosctl apigateway gateway list -o json 2> /dev/null
+    assert_success
+    assert_output -p "\"name\": \"cli-test-gateway\""
+
+    # List Gateway (Column output)
+    run ionosctl apigateway gateway list --cols name --no-headers
+    assert_success
+    assert_output -p "cli-test-gateway"
+}
+
+@test "Get ApiGateway Gateways" {
+    gateway_id=$(cat /tmp/bats_test/gateway_id)
+
+    # Get Gateway (JSON output)
+    run ionosctl apigateway gateway get --gateway-id "$gateway_id" -o json 2> /dev/null
+    assert_success
+    assert_output -p "\"status\": \"AVAILABLE\""
+}
+
+@test "Create ApiGateway Route" {
+    gateway_id=$(cat /tmp/bats_test/gateway_id)
+    route_name="route"
+    run ionosctl apigateway route create --gateway-id "$gateway_id" --name "name_route" --paths "path_route" --methods "GET" --host "host.com" -o json 2> /dev/null
+    assert_success
+
+    route_id=$(echo "$output" | jq -r '.id')
+
+
+    # Verify specific fields
+    assert_output -p "\"name\": \"name_route\""
+    assert_output -p "\"path_route\""
+    assert_output -p "\"GET\""
```

Comment:
these two checks lack the verified key, they just check the existance of that value in the JSON